pinot-transport/src/test/java/com/linkedin/pinot/transport/common/routing/RoutingTableTest.java

```diff
@@ -101,6 +106,47 @@ public void testHelixExternalViewBasedRoutingTable() throws Exception {
     }
   }
 
+  @Test
+  public void testTimeBoundaryRegression() throws Exception {
+    final FakePropertyStore propertyStore = new FakePropertyStore();
+    final OfflineSegmentZKMetadata offlineSegmentZKMetadata = new OfflineSegmentZKMetadata();
+    offlineSegmentZKMetadata.setTimeUnit(TimeUnit.DAYS);
+    offlineSegmentZKMetadata.setEndTime(1234L);
+
+    propertyStore.setContents(ZKMetadataProvider.constructPropertyStorePathForSegment("myTable_OFFLINE",
+        "someSegment_0"), offlineSegmentZKMetadata.toZNRecord());
+
+    final ExternalView offlineExternalView = new ExternalView("myTable_OFFLINE");
+    offlineExternalView.setState("someSegment_0", "Server_1.2.3.4_1234", "ONLINE");
+
+    final MutableBoolean timeBoundaryUpdated = new MutableBoolean(false);
+
+    HelixExternalViewBasedRouting routingTable = new HelixExternalViewBasedRouting(propertyStore, NO_LLC_ROUTING, null) {
+      @Override
+      protected ExternalView fetchExternalView(String table) {
+        return offlineExternalView;
+      }
+
+      @Override
+      protected void updateTimeBoundary(String tableName, ExternalView externalView) {
+        if (tableName.equals("myTable_OFFLINE")) {
+          timeBoundaryUpdated.setValue(true);
+        }
+      }
+    };
+    routingTable.setBrokerMetrics(new BrokerMetrics(new MetricsRegistry()));
+
+    Assert.assertFalse(timeBoundaryUpdated.booleanValue());
+
+    final ArrayList<InstanceConfig> instanceConfigList = new ArrayList<>();
+    instanceConfigList.add(new InstanceConfig("Server_1.2.3.4_1234"));
+
+    routingTable.markDataResourceOnline("myTable_OFFLINE", offlineExternalView, instanceConfigList);
+    routingTable.markDataResourceOnline("myTable_REALTIME", new ExternalView("myTable_REALTIME"), null);
+
+    Assert.assertTrue(timeBoundaryUpdated.booleanValue());
```

Comment:
This test is only to test for this specific regression, the routing table being built properly is checked as a part of the integration tests.